### ISSUE TITLE
QuarkusTestExtension#runExtensionMethod() should only deep clone when necessary

### ISSUE THREAD

**Issue by knutwannheden** [quarkusio/quarkus#10152]
**Description**
`QuarkusTestExtension#runExtensionMethod()` appears to do a deep-clone of all arguments of a given test method. If the parameter class is the same in both class loaders (i.e. both loaded it through a common ancestor class loader) this wouldn't be necessary and avoiding the deep-clone would also be desirable because not all objects can be cloned (see #9677).

**Comment by geoand** [comment]
Unfortunately it's not that simple. If for example you have a `Supplier` that supplies an object that was created from the System classloader, you still need to clone the  supplied object into the TCCL.

Now all this stuff is definitely nasty, I don't see any good solution to the problem.

**Comment by knutwannheden** [comment]
Yes, of course one of the deeply referenced objects may need cloning. So the cloning algorithm would then first have to check if any of the objects in the graph needs cloning or not, which isn't very nice either...

**Comment by knutwannheden** [comment]
Possibly it would help the JUnit team if the Quarkus use case could be explained in detail in the context of junit-team/junit5#201.

**Comment by geoand** [comment]
@stuartwdouglas has already chatted with the JUnit team about this, so they are aware of our case

**Comment by knutwannheden** [comment]
Unless you will consider implementing a more clever deep cloning algorithm (only cloning if at least one object in the graph needs cloning), I think we can close this issue again. Hopefully the JUnit 5 issue will be addressed in the not too distant future...

**Comment by geoand** [comment]
Hopefully we won't have to create a more clever cloning algorithm, but it may come to that at some point...
